lib/AL_USDMaya/AL/usdmaya/Global.cpp

```diff
@@ -98,6 +99,56 @@ AL::event::CallbackId Global::m_preRead;
 AL::event::CallbackId Global::m_postRead;
 AL::event::CallbackId Global::m_fileNew;
 
+//----------------------------------------------------------------------------------------------------------------------
+//class of MObjects
+MSelectionList selected;
+
+//Store the current selection list, but dont store AL_USD proxies
+static void storeSelection()
+{
+  MGlobal::displayInfo("storeSelection()");
+  //set "selected" to the current selection list
+  MGlobal::getActiveSelectionList(selected);
+  for( int i=0; i<selected.length(); ++i )
+  {
+    MObject obj;
+    selected.getDependNode(i,obj);
+    MFnDependencyNode fnParent(obj);
+    // Remove if type is AL_usdmaya_Transform
+    if (fnParent.typeName() == "AL_usdmaya_Transform")
+    {
+      MGlobal::unselectByName(fnParent.name().asChar());
+    }
+    MFnDagNode fnDagNode(obj);
+    // Unselect nodes which have AL_usdmaya_ProxyShape as a child
+    for( int i=0; i!=fnDagNode.childCount(); ++i ) {
+      MObject obj = fnDagNode.child(i);
+      MFnDagNode fnChild(obj);
+      if (fnChild.typeName() == "AL_usdmaya_ProxyShape")
+      {
+        MGlobal::unselectByName(fnParent.name().asChar());
+        MGlobal::unselectByName(fnChild.name().asChar());
+      }
+    }
+  }
+  //Reset selection list after removal of AL proxies
+  MGlobal::getActiveSelectionList(selected);
+}
+
+//Reselect the selection stored in storeSelection()
+static void restoreSelection()
+{
+  MGlobal::displayInfo("restoreSelection()");
+  // iterate through the list of items set by storeSelection()
+  for( int i=0; i<selected.length(); ++i )
+  {
+    MObject obj;
+    selected.getDependNode(i,obj);
+    MFnDependencyNode fn(obj);
+    MGlobal::selectByName(fn.name().asChar());
+  }
```

Comment:
Is there any reason you are not using ```MGlobal::setActiveSelectionList(selected);``` here? This method would fail for selected instances...